6/README.md

```diff
@@ -205,45 +228,110 @@ The endpoint must accept
+   This may include the following OpenAPI schema data types:
+        object
+        array
+        integer
+        number
+        string
+        boolean
+        
+- The parameters may be used to display swagger interface documentation.
 
 #### Design considerations
 
 This section is non-normative.
 
-The values are categorised into two types (`asset` and `json`) to support libraries such as Starfish in:
+* The `string` data type mave have the extended fields:
+    - x-resolve
+    - x-format
+
+
+* The **x-resolve** and **x-format** must have one of the following values:
+    - did-asset
+    - did
+
+- **did-asset** describes the type asset as a did/asset. For example
+
+
+```
+    # full asset id
+    
+    did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea/0xf8c12e287be8199a8c12eb9bbb63bc302c318b2290ef1e40af5a934e3315022d
+```
+
+- **did** describes a single did agent. For example
+
+```
+    # did resolve to an agent
+    
+    did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea
+    
+```
+
+- **x-resolve** Trys to resolve the asset id or did depending on the reference value.
+
+- **x-format** Validates the string value to make sure the string value is the correct format ( did or did-asset)
 
-- Validating if payloads adhere to the operation schema 
-- Adding support for metadata such as `assets` which require `did`s and `access_token`s to be fully specified.
 
 Here's an example of an request that defines a single input asset of type asset. The operation accepts an asset as input and returns the hash of the asset.
 
 - The key `to_hash` is the parameter name required by the operation (as declared in the operation's metadata)
-- Since the type is `asset` (as declared in the schema), the value must be a map with the `did` (and other optional keys)
+- Since the type is `string` with the `x-format` set to `did-asset` (as declared in the schema), the value must be a full asset address with did and asset id.
 
 ```
 {
     "to_hash": {
-             "did" : "did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea"
+         "asset_id" : {
+            "type": "string",
+            "x-format": "did-asset"
+        }
     }
 }
+
+# calling
+
+{
+    "to_hash": {
+         "asset_id" : "did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea/0xf8c12e287be8199a8c12eb9bbb63bc302c318b2290ef1e40af5a934e3315022d"
+    }
+
+}
 ```
 
-Here's an example of a request to the same operation which includes an optional parameter, the algorithm to be used for computing the hash.
+Here's an example of a request anoher operation that requires just the agent which also includes an optional parameter, the algorithm to be used for computing the hash.
 
 ```
 {
     "to_sign": {
-             "did" : "did:op:4d517500da0acb0d65a716f61330969334630363ce4a6a9d39691026ac7908ea"
+         "did" : {
+            "type": "string",
```

Comment:
Why would someone want to send a DID as a String? Why not just use a parameter of type `asset`?